src/app/auth/profiles/profiles.component.ts

```diff
@@ -0,0 +1,128 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { AppConfig } from '../../../environments/environment';
+import { MatDialog } from '@angular/material';
+import {
+  JsonParseErrorDialogComponent
+} from './json-parse-error-dialog/json-parse-error-dialog.component';
+import ProfilesJson from './profiles.json';
+
+/**
+ * Indicates all the elements that make up a Profile.
+ */
+export interface Profile {
+  profileName: string;
+  username: string;
+  password: string;
+  encodedText: string;
+}
+
+/**
+ * This component is responsible allowing the user to select a Profile
+ * in the authentication page.
+ */
+
+@Component({
+  selector: 'app-profiles',
+  templateUrl: './profiles.component.html',
+  styleUrls: ['./profiles.component.css']
+})
+export class ProfilesComponent implements OnInit {
+
+  profiles: Profile[] = undefined; // List of profiles taken from profiles.json
+  blankProfile: Profile = {profileName: '', password: '', username: '', encodedText: ''}; // A blank profile to reset values
+
+  private readonly fs = require('fs');
+
+  // path of profile.json (same folder as app for production)
+  private filePath: string = __dirname.split('CATcher', 1)[0].concat('profiles.json');
+
+  @Output() selectedProfile: EventEmitter<Profile> = new EventEmitter<Profile>();
+
+  constructor(public errorDialog: MatDialog) { }
+
+  ngOnInit() {
+
+    // Developer Mode indicates that the profiles.json is internal.
+    if (this.isDeveloperMode()) {
+
+      this.profiles = ProfilesJson.profiles;
+
+      setTimeout(() => {
+        if (!this.isValid(this.profiles)) {
+          this.openErrorDialog();
+          this.profiles = undefined;
+        }
+      });
+
+    // In Production mode the profiles.json is sourced from same folder as app.
+    } else {
+
+      if (this.userProfileFileExists(this.filePath)) {
+
+        try {
+          this.profiles = JSON.parse(
+            this.fs.readFileSync(this.filePath))['profiles'];
+        } catch (e) {
+          console.log(e);
+        }
+
+        setTimeout(() => {
+          if (!this.isValid(this.profiles)) {
+            this.openErrorDialog();
+            this.profiles = undefined;
+          }
+        });
+      }
+      // If profiles.json is not in the same folder as the app, do nothing.
+    }
+  }
+
+  /**
+   * Makes Error dialog visible to the user.
+   */
+  openErrorDialog(): void {
+    this.errorDialog.open(JsonParseErrorDialogComponent);
+  }
+
+  /**
+   * Checks that every profile is correctly defined in the array of profiles.
+   * @param profiles - Array of profiles sourced from profiles.json
+   * @return true if all profiles are correctly defined, false if otherwise.
```

Comment:
The `@return` can be omitted as it just repeats line 88?

Reply:
Yes